internal/benchrunner/runners/system/runner.go

```diff
@@ -691,6 +708,159 @@ func (r *runner) enrollAgents() error {
+	}
+
+	var mapping string
+	for _, v := range mappings {
+		mapping = string(v.Mappings)
+	}
+
+	reader := bytes.NewReader(
+		[]byte(fmt.Sprintf(`{
+			"settings": {"number_of_replicas":0},
+			"mappings": %s
+		}`, mapping)),
+	)
+
+	indexName := fmt.Sprintf("bench-reindex-%s-%s", r.runtimeDataStream, r.ctxt.Test.RunID)
+
+	logger.Debugf("creating %s index in metricstore...", indexName)
+
+	createRes, err := r.options.ESMetricsAPI.Indices.Create(
+		indexName,
+		r.options.ESMetricsAPI.Indices.Create.WithBody(reader),
+	)
+	if err != nil {
+		return fmt.Errorf("could not create index: %w", err)
+	}
+	defer createRes.Body.Close()
+
+	if createRes.IsError() {
+		return errors.New("got a response error while creating index")
+	}
+
+	bodyReader := strings.NewReader(`{"query":{"match_all":{}}}`)
+
+	logger.Debug("starting scrolling of events...")
+	res, err := r.options.ESAPI.Search(
+		r.options.ESAPI.Search.WithIndex(r.runtimeDataStream),
+		r.options.ESAPI.Search.WithBody(bodyReader),
+		r.options.ESAPI.Search.WithScroll(time.Minute),
+		r.options.ESAPI.Search.WithSize(10000),
+	)
+	if err != nil {
+		return fmt.Errorf("error executing search: %w", err)
+	}
+	defer res.Body.Close()
+
+	// Iterate through the search results using the Scroll API
+	for {
+		var sr map[string]interface{}
+		if err := json.NewDecoder(res.Body).Decode(&sr); err != nil {
+			return fmt.Errorf("error decoding search response: %w", err)
+		}
+
+		resErr, found := sr["error"]
+		if found {
+			errStr := resErr.(map[string]interface{})["reason"].(string)
+			return fmt.Errorf("error searching for documents: %s", errStr)
+		}
+
+		hits, found := sr["hits"].(map[string]interface{})["hits"].([]interface{})
+		if !found || len(hits) == 0 {
+			break
+		}
+
+		var bulkBodyBuilder strings.Builder
+		for _, hit := range hits {
+			bulkBodyBuilder.WriteString(fmt.Sprintf("{\"index\":{\"_index\":\"%s\",\"_id\":\"%s\"}}\n", indexName, hit.(map[string]interface{})["_id"]))
+			enriched := r.enrichEventWithBenchmarkMetadata(hit.(map[string]interface{})["_source"].(map[string]interface{}))
+			src, err := json.Marshal(enriched)
+			if err != nil {
+				return fmt.Errorf("error decoding _source: %w", err)
+			}
+			bulkBodyBuilder.WriteString(fmt.Sprintf("%s\n", string(src)))
+		}
+
+		logger.Debugf("bulk request of %d events...", len(hits))
+
+		bulkRes, err := r.options.ESMetricsAPI.Bulk(strings.NewReader(bulkBodyBuilder.String()))
+		if err != nil {
+			return fmt.Errorf("error performing the bulk index request: %w", err)
+		}
+		bulkRes.Body.Close()
+
+		scrollId, found := sr["_scroll_id"].(string)
+		if !found {
+			return errors.New("error getting scroll ID")
+		}
+
+		res, err = r.options.ESAPI.Scroll(
+			r.options.ESAPI.Scroll.WithScrollID(scrollId),
+			r.options.ESAPI.Scroll.WithScroll(time.Minute),
+		)
+		if err != nil {
+			return fmt.Errorf("error executing scroll: %s", err)
+		}
+		defer res.Body.Close()
```

Comment:
Should this be executed at this moment instead of defering it at the end of the function ? 

```suggestion
		res.Body.Close()
```